streaming/src/main/scala/org/apache/spark/streaming/dstream/DStream.scala

```diff
@@ -980,6 +985,40 @@ abstract class DStream[T: ClassTag] (
     ssc.graph.addOutputStream(this)
     this
   }
+
+
+  /**
+   * Same as initialize method but not initializing dependencies
+   * @param time
+   */
+  def initializeAfterContextStart(time: Time) {
+    if (!isInitialized) {
+      if (zeroTime != null && zeroTime != time) {
+        throw new SparkException("ZeroTime is already initialized to " + zeroTime
+            + ", cannot initialize it again to " + time)
+      }
+      zeroTime = time
+
+      // Set the checkpoint interval to be slideDuration or 10 seconds, which ever is larger
+      if (mustCheckpoint && checkpointDuration == null) {
+        checkpointDuration = slideDuration * math.ceil(Seconds(10) / slideDuration).toInt
+        logInfo("Checkpoint interval automatically set to " + checkpointDuration)
+      }
+
+      // Set the minimum value of the rememberDuration if not already set
+      var minRememberDuration = slideDuration
+      if (checkpointDuration != null && minRememberDuration <= checkpointDuration) {
+        // times 2 just to be sure that the latest checkpoint is not forgotten (#paranoia)
+        minRememberDuration = checkpointDuration * 2
+      }
+      if (rememberDuration == null || rememberDuration < minRememberDuration) {
+        rememberDuration = minRememberDuration
+      }
+
+      // Initialize the dependencies
+      dependencies.foreach(_.initializeAfterContextStart(zeroTime))
+    }
+  }
```

Comment:
This is almost an exact copy of other initialize (which was changed in upstream for Spark 2.0). What I had done was pass a "skipInitialized" flag to other initialize method with minimal changes for that. That is a more maintainable option because when initialize again changes upstream, then we do not need to manually update initializeAfterContextStart.

Can you compare the code of this method and initialize(time, skipInitialized = true) and see what is causing the problem? We should make that fix in initialize instead.